source/extensions/filters/http/composite/filter.h

```diff
@@ -25,6 +28,22 @@ struct FilterStats {
   ALL_COMPOSITE_FILTER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
+class MatchedActionInfoType : public StreamInfo::FilterState::Object {
+public:
+  MatchedActionInfoType() {}
+
+  ProtobufTypes::MessagePtr serializeAsProto() const override { return buildProtoStruct(); }
+
+  absl::optional<std::string> serializeAsString() const override;
```

Comment:
1 . The ACTION_NAME is the filter name to be apply
```
matchers:
                          - predicate:
                              single_predicate:
                                input:
                                  name: request-headers
                                  typed_config:
                                    "@type": type.googleapis.com/envoy.type.matcher.v3.HttpRequestHeaderMatchInput
                                    header_name: h1
                                value_match:
                                  safe_regex:
                                    google_re2: {}
                                    regex: value2
                            on_match:
                              action:
                                name: execute-filter-action-level-00002
                                typed_config:
                                  "@type": type.googleapis.com/envoy.extensions.filters.http.composite.v3.ExecuteFilterAction
                                  typed_config:
                                    name: **ACTION_NAME**
                                    typed_config:
                                      "@type": type.googleapis.com/envoy.extensions.filters.http.ext_proc.v3.ExternalProcessor
                                      grpc_service:
                                        envoy_grpc:
                                          cluster_name: httpbin-ext-authz-native-external-proc
                                        timeout: 5s
                                      processing_mode:
                                        request_header_mode: SEND
                                        response_header_mode: SEND
                                        request_trailer_mode: SKIP
                                        response_trailer_mode: SKIP
                                      message_timeout: 2s
```

2 - Yes, it can be consumed with access log and from a custom WASM on_log method, in this particular case the stringlified value is used, that is the reason to convert to json

```
access_log:
          - name: envoy.access_loggers.file
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
              path: "/dev/stdout"
              typed_json_format: 
                status: "%RESPONSE_CODE%"
                jmo2: "%FILTER_STATE(envoy.extensions.filters.http.composite.matched_actions)%"
                jmo3: "%FILTER_STATE(envoy.filters.network.http_connection_manager.local_reply_owner:PLAIN)%"
```

Reply:
Thanks for details!

Yea, there was a typo in config example in my comment. Using this [name](https://github.com/envoyproxy/envoy/blob/7e834d7123f38e4f607b8fa94b13167f0992de1f/api/envoy/config/core/v3/extension.proto#L23) as ACTION_NAME sounds good to me. 

This is actually the name of embedded filter.

Reply:
Excelent, I have added 2 more test to check the 2 types of config, dynamic and typed after the merge from main